vignettes/civis_ml.Rmd

```diff
@@ -0,0 +1,300 @@
+---
+title: "Machine Learning in R with CivisML"
+author: "Patrick Miller"
+date: "2017-08-30"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{civis_ml}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+There are so many models to build! When this becomes challenging on a local machine, offloading model building to the cloud can save a lot of time and effort.
+
+[CivisML](https://www.civisanalytics.com/blog/civisml-scikit-learn-at-scale/) is a machine learning service on Civis Platform that makes this as painless as possible. You can fit many different models, do extensive metaparameter tuning, and score data sets with millions of observations stored in remote databases. Once these models are built, they live in Civis Platform permanently and can be included into production pipelines. Results can be easily incorporated into reports and dashboards.
+
+CivisML is built in Python using [scikit-learn](http://scikit-learn.org/stable/), and leverages AWS behind the scenes for efficient distributed computing. However, most of its features can be used through R without knowledge of Python or AWS with the `civis_ml` function in `civis`.
+
+While `civis_ml` is a complex function with many arguments, basic machine learning modeling and scoring can be easily carried out easily. We illustrate several features of `civis_ml` with data from a fictious company called Brandable, who is looking to predict which customers are likely to upgrade from the free to the premium service.
+
+## Data sources
+
+The first step of modeling with `civis_ml` is to specify the data source, which is the first argument. `civis_ml` works with local data frames, a CSV on local disk, tables in Redshift, and files on S3 (the files endpoint):
+
+```{r, eval = FALSE}
+library(civis)
+civis_ml(df, ...)
+civis_ml("path/to/data.csv", ...)
+civis_ml(civis_table(table_name = "schema.table", database_name = "database"), ...)
+civis_ml(civis_file(1234), ...)
+```
+
+The Brandable data is located in a Redshift table called `sample_project.premium_training_set`:
+
+```{r, eval  = FALSE}
+tab <- civis_table(table_name = "sample_project.premium_training_set", 
+                   database_name = "redshift-general")
```

Comment:
Just want to make sure my other comment on `redshift-general` isn't lost.  This database name will likely be unfamiliar to most users.